tests: ztest: fixed off-by-one in sys_bitfield_find_first_clear

mtpr-ot opened the pull request:
Calling __builtin_ffsl(neg_bitmap) returns first bit set in the word,
e.g. 4 if bitmap is 11111000. As this must translate to zero-based index
3, one must be subtracted from the result.

Signed-off-by: Morten Priess <mtpr@oticon.com>

carlescufi commented:
@mtpr-ot can you split the changes to `ztest_mock.c` into 2 commits? one for the indexing fix, the other for the addition of the function

SebastianBoe commented on tests/ztest/test/mock/src/main.c:
```diff
@@ -40,25 +40,25 @@ static void return_value_tests(void)
 			 );
 
 	ztest_run_test_suite(mock_framework_tests);
-}
```
This commit must be squashed into the first commit.

mtpr-ot committed:
tests: ztest: fixed off-by-one in sys_bitfield_find_first_clear

Calling __builtin_ffsl(neg_bitmap) returns first bit set in the word,
e.g. 4 if bitmap is 11111000. As this must translate to zero-based index
3, one must be subtracted from the result.

Signed-off-by: Morten Priess <mtpr@oticon.com>

mtpr-ot committed:
tests: ztest: added test case for multiple mock expects

The test verifies that the mock framework can handle functions
under test which call more than one mock function, and
correctly processes each call.

Signed-off-by: Morten Priess <mtpr@oticon.com>
